### PR TITLE
#168781807 setup social login

### PR DESCRIPTION
**Pull request by Crispy-rw**
#### What does this PR do?
 Integrate login with social media account(facebook,google +)
#### Description of Task to be completed?

 This PR allows to create and sign in using social media account like facebook and google so 
 whenever you have facebook or google account you can use them to access our application 
 with a verified email address

#### How should this be manually tested?

- clone this repo go in the project directory

- run this command `npm i` to install all dependencies

- run this command `npm run dev` in your command line

- In Postman, create a new GET request and enter the following URL: 
 
2.  for facebook POST method `localhost:3000/api/v1/users/auth/facebook`  and the bod

1. for google  POST method`localhost:3000/api/v1/users/auth/google` .

3. Add the following environment variables to your application
   `GOOGLE_CLIENT_ID=`
   `GOOGLE_CLIENT_SECRET=`
   `FACEBOOK_CLIENT_ID=`
   `FACEBOOK_CLIENT_SECRET=`
4. Go to the following url https://developers.google.com/oauthplayground/ and authorize the 
      following APIs (This is done in step 1 of the Playground process).
       a) Google OAuth apiv2 and choose 

- https://www.googleapis.com/auth/userinfo.email

- https://www.googleapis.com/auth/userinfo.profile
And follows the all the remaining process
5. Create an app on  [facebook developer site](https://developers.facebook.com) and create a user access token Or you can be added as a developer on my app use my  keys

#### Any background context you want to provide?
  N/A

#### What are the relevant pivotal tracker stories?
 #168781807

### Screenshoot

Screen Shot 2019-10-14 at 17 12 35
Screen Shot 2019-10-14 at 17 12 45

### CODE REVIEW

File: src/config/auth.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  facebook: {
+    clientID: '2618632188182397',
+    clientSecret: 'f8a48d72223691ad05ca936fa20049e9',
+  },
+  google: {
+    clientID: '421632939170-a4an5ufi14580qmt5ll9d7r7bkesjmfu.apps.googleusercontent.com',
+    clientSecret: '1LysFi5OOg1DfjyVRhXQIIgH',
+  }
```
Comment (urbanishimwe) [inline src/config/auth.js]: you need to hide those details, just send them in slack

**Review by urbanishimwe — CHANGES_REQUESTED** [review]

File: src/config/auth.js (hunk 1 above)
Comment (urbanishimwe) [reply src/config/auth.js]: and well done @Crispy-lang

**Review by lemoissonM — CHANGES_REQUESTED** [review]
there is no documentation for this feature, please add one

**Comment by Crispy-rw** [comment]
> there is no documentation for this feature, please add one

Yaaa I added the documentation Please check

**Review by alexismajya — APPROVED** [review]

**Review by urbanishimwe — APPROVED** [review]

**Review by swaibat — APPROVED** [review]

**Review by lemoissonM — APPROVED** [review]

File: src/controllers/userController.js [hunk 2]
```diff
@@ -64,5 +64,19 @@ const User = {
...
+    const [data] = await db.Users.findOrCreate({
+      where: { email: req.user.email },
+      defaults: req.user
+    });
+
+    return sendResult(res, 201, 'User logged successfully', {
+      userId: data.id,
+      username: data.username,
+      email: data.email,
+      token: helpers.createToken(data.id, data.email, data.isverified)
```
Comment (nakiwuge) [inline src/controllers/userController.js]: Destructure `id`, `email`,`isverified` and `username` from data

File: src/config/passport.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+
+import FacebookTokenStrategy from 'passport-facebook-token';
+import GooglePlusTokenStrategy from 'passport-google-plus-token';
+import passport from 'passport';
+import config from './auth';
+import OAuthCallback from '../utils/OAuthCallback';
+
```
Comment (nakiwuge) [inline src/config/passport.js:7]: Remove this line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@Crispy-lang great job kindly check out the comments I have left on this pr.

File: src/controllers/userController.js (hunk 2 above)
Comment (Crispy-rw) [reply src/controllers/userController.js]: I corrected it

File: src/config/passport.js (hunk 3 above)
Comment (Crispy-rw) [reply src/config/passport.js:7]: I removed it

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**